torch/distributed/fsdp/_fully_shard/_fsdp_collectives.py

```diff
@@ -1,3 +1,5 @@
+# mypy: disable-error-code=possibly-undefined
```

Comment:
I think just caching world_size > 1 as a Boolean might fix this error code. Also better to have this inline type ignore  right?